docs/configuration.md

```diff
@@ -37,7 +37,7 @@ directory.
 >as replacing the storage driver type with `REGISTRY_STORAGE=filesystem`, then
 >all sub-fields will be erased. As such, specifying the storage type in the
 >environment will remove all parameters related to the old storage
->configuration.
+>configuration (order *matters*).
```

Comment:
"order matters": I noticed that `-e FOO=bar -e FOO_FOO=bar` does not have the same effect as `-e FOO_FOO=bar -e FOO=foo`.

If your PR fixes the oddities with env vars, then yeah, this will soon not matter anymore.